File: chap3/Factorielle.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)
let _ =
  let n = int_of_string (input_line stdin) in
  let f = ref 1 in
  for i = 1 to n do
   f := !f * i
  done;
  print_int !f; print_newline ()

File: chap3/Factorielle.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>
#include <iomanip>
#include <iostream>

using namespace std;

int main() 
{
   int n, i, f;
   cin >> n;
   f = 1;
   for (i = 1; i <= n; i = i + 1) 
   {
      f = f * i;
   }
   cout << f << endl;
}
